pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractReplicator.java

```diff
@@ -188,58 +274,134 @@ protected CompletableFuture<Boolean> isLocalTopicActive() {
         }, brokerService.executor());
     }
 
-    protected synchronized CompletableFuture<Void> closeProducerAsync() {
-        if (producer == null) {
-            STATE_UPDATER.set(this, State.Stopped);
+    /**
+     * @Deprecated This method only be used by {@link PersistentTopic#checkGC} now.
+     * TODO "PersistentReplicator.replicateEntries" may get a NullPointerException if this method and a
+     *   "cursor.readComplete" execute concurrently.
+     */
+    @Deprecated
```

Comment:
If it still be used in PersistentTopic#checkGC, I don't think we should deprecate this method. Do we have any other alternative for `PersistentTopic#checkGC`?

Reply:
I think we can left a TODO there to refactor the code.

Reply:
`@Deprecated` has been removed.

Reply:
> I think we can left a TODO there to refactor the code.

Removed todo these comments, and I left another TODO comment [here](https://github.com/apache/pulsar/pull/21946/files#diff-5edf14cc6f25857d0cfdd26b2d3b3141230ecfb0dfa95aebf7583fd76ede4c4bR2944-R2952)